packages/ag-charts-community/src/chart/axis/axis.ts

```diff
@@ -1069,6 +1071,18 @@ export abstract class Axis<S extends Scale<D, number, TickInterval<S>> = Scale<a
             const rawTick = rawTicks[i];
             const translationY = scale.convert(rawTick) + halfBandwidth;
 
+            // Do not render ticks outside the series rect. A clip rect would trim long labels, so instead hide ticks
+            // based on their translation.
+            if (seriesRect) {
+                if (this.direction === ChartAxisDirection.X && (translationY < 0 || translationY > seriesRect.width)) {
+                    continue;
+                }
+
+                if (this.direction === ChartAxisDirection.Y && (translationY < 0 || translationY > seriesRect.height)) {
+                    continue;
+                }
+            }
+
```

Comment:
Is this going to play well with animations?

We might want to defer this culling until the animation processing, somewhere around here?
https://github.com/ag-grid/ag-charts/blob/1066f9ada5083f31f3d82c5c448cdb21d154a5e6/packages/ag-charts-community/src/chart/axis/axisUtil.ts#L60-L63

Reply:
This all appears to work with animation, including adding nodes before and after and with ticking updates.